ibacm/src/acme.c

```diff
@@ -980,8 +980,12 @@ static int query_svcs(void)
 	for (i = 0; svc_list[i]; i++) {
 		ret = ib_acm_connect(svc_list[i]);
 		if (ret) {
-			printf("%s,unable to contact service: %s\n",
-				svc_list[i], strerror(errno));
+			if (ret == -1)
```

Comment:
What is the semantic difference between -1 and errno?  It seems like -1 is just an error without any "meaning"?  Why?

Reply:
Because getaddrinfo() in ib_acm_connect() does not always set errno on failure.